src/main/java/com/limechain/teavm/HttpRequest.java

```diff
@@ -1,8 +1,32 @@
 package com.limechain.teavm;
 
+import org.teavm.interop.Async;
+import org.teavm.interop.AsyncCallback;
 import org.teavm.jso.JSBody;
+import org.teavm.jso.JSFunctor;
+import org.teavm.jso.JSObject;
+import org.teavm.jso.core.JSError;
+
+import java.io.IOException;
 
 public class HttpRequest {
-    @JSBody(params = {"method", "url", "body"}, script = "return httpRequestSync(method, url, body);")
-    public static native String httpRequestSync(String method, String url, String body);
+    @JSFunctor
+    interface HttpRequestCallback extends JSObject {
+        void apply(JSError error, String response);
+    }
+
+    @Async
+    public static native String asyncHttpRequest(String method, String url, JSObject body);
+    private static void asyncHttpRequest(String method, String url, JSObject body, AsyncCallback<String> callback) {
```

Comment:
```suggestion
    public static native String asyncHttpRequest(String method, String url, String body);
    
    private static void asyncHttpRequest(String method, String url, String body, AsyncCallback<String> callback) {
```

I know these 2 methods are related, but I suggest we conform to the rule of having empty rows between methods.
Also regarding the JSObject param see [this](https://github.com/LimeChain/Fruzhin-Web/pull/4#issuecomment-2287066791) comment.